swarm/network/stream/stream.go

```diff
@@ -403,6 +310,11 @@ func (r *Registry) setPeer(peer *Peer) {
 	r.peersMu.Lock()
 	r.peers[peer.ID()] = peer
 	metrics.GetOrRegisterGauge("registry.peers", nil).Update(int64(len(r.peers)))
+
```

Comment:
this should not be hidden in `setPeer` but explicitly part of `Run` function

Reply:
Yes, I agree, it is correct to have it in Run function. Changed.